pkg/diagnostics/client/run_diagnostics_pod.go

```diff
@@ -0,0 +1,163 @@
+package client
+
+import (
+	"bufio"
+	"fmt"
+	"regexp"
+	"strconv"
+	"time"
+
+	kapi "k8s.io/kubernetes/pkg/api"
+	kclient "k8s.io/kubernetes/pkg/client/unversioned"
+
+	osclientcmd "github.com/openshift/origin/pkg/cmd/util/clientcmd"
+	"github.com/openshift/origin/pkg/cmd/util/variable"
+	"github.com/openshift/origin/pkg/diagnostics/types"
+)
+
+const (
+	DiagnosticPodName = "DiagnosticPod"
+)
+
+// DiagnosticPod is a diagnostic that runs a diagnostic pod and relays the results.
+type DiagnosticPod struct {
+	KubeClient          kclient.Client
+	Namespace           string
+	Level               int
+	Factory             *osclientcmd.Factory
+	PreventModification bool
+	ImageTemplate       variable.ImageTemplate
+}
+
+// Name is part of the Diagnostic interface and just returns name.
+func (d *DiagnosticPod) Name() string {
+	return DiagnosticPodName
+}
+
+// Description is part of the Diagnostic interface and provides a user-focused description of what the diagnostic does.
+func (d *DiagnosticPod) Description() string {
+	return "Create a pod to run diagnostics from the application standpoint"
+}
+
+// CanRun is part of the Diagnostic interface; it determines if the conditions are right to run this diagnostic.
+func (d *DiagnosticPod) CanRun() (bool, error) {
+	if d.PreventModification {
+		return false, fmt.Errorf("Running the diagnostic pod is an API change, which is prevented as you indicated.")
+	} else {
+		return true, nil
+	}
+}
+
+// Check is part of the Diagnostic interface; it runs the actual diagnostic logic
+func (d *DiagnosticPod) Check() types.DiagnosticResult {
+	r := types.NewDiagnosticResult("DiagnosticPod")
+	d.runDiagnosticPod(nil, r)
+	return r
+}
+
+func (d *DiagnosticPod) runDiagnosticPod(service *kapi.Service, r types.DiagnosticResult) {
+	loglevel := d.Level
+	if loglevel > 2 {
+		loglevel = 2 // need to show summary at least
+	}
+	imageName := d.ImageTemplate.ExpandOrDie("deployer")
+	pod, err := d.KubeClient.Pods(d.Namespace).Create(&kapi.Pod{
+		ObjectMeta: kapi.ObjectMeta{GenerateName: "pod-diagnostic-test-"},
+		Spec: kapi.PodSpec{
+			RestartPolicy: kapi.RestartPolicyNever,
+			Containers: []kapi.Container{
+				{
+					Name:    "pod-diagnostics",
+					Image:   imageName,
+					Command: []string{"pod-diagnostics", "-l", strconv.Itoa(loglevel)},
+				},
+			},
+		},
+	})
+	if err != nil {
+		r.Error("DCli2001", err, fmt.Sprintf("Creating diagnostic pod with image %s failed. Error: (%[2]T) %[2]v", imageName, err))
+		return
+	}
+	defer func() { // delete what we created, or notify that we couldn't
```

Comment:
Yes. The logging code is fragile so I was thinking that in case of an unexpected error where my logs didn't show up on the spot, I would love to have the pod around and try `oc logs` instead of re-running the command.